src/main/java/picard/illumina/ExtractIlluminaBarcodes.java

```diff
@@ -255,27 +171,31 @@ protected int doWork() {
         LOG.info("Processing with " + numProcessors + " PerTileBarcodeExtractor(s).");
         final ThreadPoolExecutor pool = new ThreadPoolExecutorWithExceptions(numProcessors);
 
+        // Create BarcodeMetric for counting reads that don't match any barcode
+        final String[] noMatchBarcode = new String[readStructure.sampleBarcodes.length()];
+        int index = 0;
+        for (final ReadDescriptor d : readStructure.descriptors) {
+            if (d.type == ReadType.Barcode) {
+                noMatchBarcode[index++] = StringUtil.repeatCharNTimes('N', d.length);
+            }
+        }
```

Comment:
This is repeated in a handful of locations - perhaps extract to a static function somewhere?